dataloaders/NGCFConverter/ngcf_converter.py

```diff
@@ -0,0 +1,236 @@
+            write_test_split(user_items_test_dict, i)
+
+
+def convert_data_yahoo(full_data_path, k_fold_path):
+    item_remapping, user_remapping = {}, {}
+    item_counter, user_counter = 0, 0
+
+    with open(full_data_path) as f:
+        for line in f.readlines():
+            if len(line) > 0:
+                line = line.split(',')
+                user_id = line[1]
+                item_id = line[2]
+
+                if item_id not in item_remapping.keys():
+                    item_remapping[item_id] = user_counter
+                    user_counter += 1
+                if user_id not in user_remapping.keys():
+                    user_remapping[user_id] = item_counter
+                    item_counter += 1
+
+    for i in range(5):
+        read_and_write_split(k_fold_path, user_remapping, item_remapping)
+
+    write_user_list(user_remapping)
+    write_item_list(item_remapping)
+
+
+def convert_data_comoda(full_data_path, k_fold_path):
+    item_remapping, user_remapping = {}, {}
+    item_counter, user_counter = 0, 0
+
+    with open(full_data_path) as f:
+        for line in f.readlines():
+            if len(line) > 0:
+                line = line.strip().split(';')
+                user_id = line[0]
+                item_id = line[1]
+
+                if item_id not in item_remapping.keys():
+                    item_remapping[item_id] = item_counter
+                    item_counter += 1
+                if user_id not in user_remapping.keys():
+                    user_remapping[user_id] = user_counter
+                    user_counter += 1
+
+    for i in range(5):
+        read_and_write_split(k_fold_path, user_remapping, item_remapping,
+                             prune=True)
+
+    write_user_list(user_remapping)
+    write_item_list(item_remapping)
+
+
+def convert_data_movielens(full_data_path, k_fold_path):
+    item_remapping, user_remapping = {}, {}
+    item_counter, user_counter = 0, 0
+
+    with open(full_data_path) as f:
+        for line in f.readlines():
+            if len(line) > 0:
+                line = line.split()
+                user_id = line[0]
+                item_id = line[1]
+
+                if item_id not in item_remapping.keys():
+                    item_remapping[item_id] = item_counter
+                    item_counter += 1
+                if user_id not in user_remapping.keys():
+                    user_remapping[user_id] = user_counter
+                    user_counter += 1
+
+    for i in range(5):
+        read_and_write_split(k_fold_path, user_remapping, item_remapping)
+
+    write_user_list(user_remapping)
+    write_item_list(item_remapping)
+
+
+def convert_data_yahoo_positive(full_data_path, k_fold_path):
+    item_remapping, user_remapping = {}, {}
+    item_counter, user_counter = 0, 0
+
+    with open(full_data_path) as f:
+        for line in f.readlines():
+            if len(line) > 0:
+                line = line.split(',')
+                if int(line[4]) >= 3:
```

Comment:
Er >= 3 for at finde relevante items. Måske det også skulle være en parameter eller der skal gives en kommentar til det

Reply:
true